.circleci/workflows.yml

```diff
@@ -0,0 +1,1151 @@
+                # have been deployed to prod. (bigquery-etl-integration-test is treated
+                # as a stage environment)
+                command: |
+                  if [ "$CIRCLE_BRANCH" = main ]; then
+                    echo "Check dry run for all queries because branch is" \
+                      "$CIRCLE_BRANCH"
+                    PATHS=sql
+                  elif git log --format=%B --no-merges -n 1 |
+                      grep -qF '[run-tests]'; then
+                    echo "Check dry run for all queries because [run-tests] in" \
+                      "commit message"
+                    PATHS=sql
+                  else
+                    PATHS="sql/bigquery-etl-integration-test"
+                  fi
+                  echo $PATHS
+                  PATH="venv/bin:$PATH" script/bqetl dryrun --validate-schemas $PATHS
+                # yamllint enable rule:line-length
+      - run: exit 0
+  validate-metadata:
+    docker: *docker
+    steps:
+      - when:
+          condition:
+            or:
+              - << pipeline.parameters.validate-sql >>
+              - << pipeline.parameters.validate-routines >>
+          steps:
+            - checkout
+            - *restore_venv_cache
+            - *build
+            - *attach_generated_sql
+            - *copy_staged_sql
+            - run:
+                name: Verify that metadata files are valid
+                command: |
+                  # TODO: Add check here to make sure all queries have metadata.yaml
+                  PATH="venv/bin:$PATH" script/bqetl query validate \
+                    --respect-dryrun-skip
+      - run: exit 0
+  integration:
+    docker: *docker
+    steps:
+      - when:
+          condition:
+            or:
+              - << pipeline.parameters.validate-bqetl >>
+          steps:
+            - checkout
+            - &skip_forked_pr
+              run:
+                name: Early return if this build is from a forked PR
+                command: |
+                  if [ -n "$CIRCLE_PR_NUMBER" ]; then
+                    echo "Cannot pass creds to forked PRs," \
+                      "so marking this step successful"
+                    circleci-agent step halt
+                  fi
+            - *restore_venv_cache
+            - *build
+            - run:
+                name: PyTest Integration Test
+                # yamllint disable rule:line-length
+                command: |
+                  PATH="venv/bin:$PATH" script/entrypoint -m 'integration' -n 8
+                # workaround for job failing with `Too long with no output (exceeded 10m0s): context deadline exceeded` error
+                no_output_timeout: 30m
+      - run: exit 0
+  generate-dags:
+    docker: *docker
+    steps:
+      - when:
+          condition:
+            or:
+              - << pipeline.parameters.validate-sql >>
+              - << pipeline.parameters.validate-routines >>
```

Comment:
- I wouldn't think that routines changing would make a difference for generating DAGs.
- The `dags.yaml` file changing should set the `validate-sql` parameter (or some other DAG-specific parameter could be used).